### PR TITLE
test(k8s-functional): cover operator#718

### PR DESCRIPTION
**Pull request by vponomaryov**
Implements functional test for https://github.com/scylladb/scylla-operator/issues/718

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: functional_tests/scylla_operator/test_functional.py [hunk 1]
```diff
@@ -356,13 +356,18 @@ def test_scylla_operator_pods(db_cluster: ScyllaPodCluster):
     assert not not_running_pods, f'There are pods in state other than running: {not_running_pods}'
 
 
-def test_startup_probe_in_scylla_pods(db_cluster: ScyllaPodCluster):
-    pods = db_cluster.k8s_cluster.kubectl('get pods -A -l "app.kubernetes.io/name=scylla" -o yaml')
+def test_startup_probe_exists_in_scylla_pods(db_cluster: ScyllaPodCluster):
+    pods = get_pods_without_probe(
+        db_cluster, "startupProbe", "app.kubernetes.io/name=scylla", "scylla")
+    assert not pods, f"startupProbe is not found in the following pods: {pods}"
```
Comment (dkropachev) [inline functional_tests/scylla_operator/test_functional.py:363]: @vponomaryov  I would guess that absense of other probes is also a problem, isn't it ? should be add them ?

**Review by dkropachev — CHANGES_REQUESTED** [review]

File: functional_tests/scylla_operator/test_functional.py (hunk 1 above)
Comment (vponomaryov) [reply functional_tests/scylla_operator/test_functional.py:363]: It is not a problem. It is optional. For the moment "readinessProbe" is the only probe in the manager pod. And the covered operator issue is exactly about it.

File: functional_tests/scylla_operator/test_functional.py [hunk 2]
```diff
@@ -356,13 +356,18 @@ def test_scylla_operator_pods(db_cluster: ScyllaPodCluster):
     assert not not_running_pods, f'There are pods in state other than running: {not_running_pods}'
 
 
-def test_startup_probe_in_scylla_pods(db_cluster: ScyllaPodCluster):
-    pods = db_cluster.k8s_cluster.kubectl('get pods -A -l "app.kubernetes.io/name=scylla" -o yaml')
+def test_startup_probe_exists_in_scylla_pods(db_cluster: ScyllaPodCluster):
+    pods = get_pods_without_probe(
```
Comment (bentsi) [inline functional_tests/scylla_operator/test_functional.py:360]: please add argument names that you call in the function

File: functional_tests/scylla_operator/test_functional.py [hunk 3]
```diff
@@ -356,13 +356,18 @@ def test_scylla_operator_pods(db_cluster: ScyllaPodCluster):
...
-                no_startup_probe_pods.append(f"pod: {pod['metadata']['name']}, container: {container['name']}")
 
-    assert not no_startup_probe_pods, f"Startup probe is not found in the pods: {no_startup_probe_pods}"
+@pytest.mark.require_mgmt()
+def test_readiness_probe_exists_in_mgmt_pods(db_cluster: ScyllaPodCluster):
+    """
+    PR: https://github.com/scylladb/scylla-operator/pull/725
+    Issue: https://github.com/scylladb/scylla-operator/issues/718
+    """
+    pods = get_pods_without_probe(
```
Comment (bentsi) [inline functional_tests/scylla_operator/test_functional.py:371]: same here, add arguments

File: functional_tests/scylla_operator/test_functional.py (hunk 2 above)
Comment (vponomaryov) [reply functional_tests/scylla_operator/test_functional.py:360]: Done

File: functional_tests/scylla_operator/test_functional.py (hunk 3 above)
Comment (vponomaryov) [reply functional_tests/scylla_operator/test_functional.py:374]: Done

**Review by dkropachev — APPROVED** [review]

**Review by bentsi — APPROVED** [review]

**Pull request merged**